chat_shell/chat_shell/tools/builtin/knowledge_listing.py

```
@@ -0,0 +1,427 @@
+    def _run(
+        self,
+        knowledge_base_id: int,
+        run_manager: CallbackManagerForToolRun | None = None,
+    ) -> str:
+        """Synchronous run - not implemented, use async version."""
+        raise NotImplementedError("KbLsTool only supports async execution")
+
+    async def _arun(
+        self,
+        knowledge_base_id: int,
+        run_manager: CallbackManagerForToolRun | None = None,
```

Ruff (0.14.14)

[warning] Unused method argument: `run_manager`

(ARG002)